### PR TITLE
initial draft of project charter; closes #4087

### LINKED ISSUE #4087: add project charter
As per OpenJS Foundation [onboarding checklist](https://github.com/openjs-foundation/cross-project-council/blob/35f0e7303520fb591c748797385779db9e6df385/proposals/stage-3/PROJECT_ONBOARDING/README.md), Mocha needs a project charter.

Here's [the project charter template](https://github.com/openjs-foundation/cross-project-council/blob/master/PROJECT_CHARTER_TEMPLATE.md) that I'll be working from.  Many of the sections are optional.

I suppose this file will live in `PROJECT_CHARTER.md` and be linked to from `README.md` and `CONTRIBUTING.md`.

Normally we'd handle stuff like this in the [admin repo](/mochajs/admin), but the file should live in _this_ repo, so here's an issue to track it.

Ref: mochajs/admin#7 and openjs-foundation/project-onboarding#27

### PR DESCRIPTION
**Pull request by boneskull**
This is a draft charter.  It's important all maintainers give their OK on this.  In addition, I think it needs to be ratified by the CPC (which I'll deal with).  But before that happens, we need to make sure it's ok.

Open to any suggestions and can hopefully answer any questions we have.  See #4087 for issue.

cc @jorydotcom

I've added the four other active maintainers to the list of reviewers, but anyone from @mocha/core (or anyone from the community, even) is welcome to offer feedback.

### CODE REVIEW

**Comment by boneskull** [comment]
We might want to take this opportunity to split [MAINTAINERS.md](https://github.com/mochajs/mocha/blob/master/MAINTAINERS.md) into multiple files (for example, pulling out part of it into a new `GOVERNANCE.md`)

**Comment by juergba** [comment]
Meanwhile I regret having agreed to putting Mocha into the "Growth" box. It's not what I'm experiencing, I feel Mocha has passed its climax and is living of its shiny past.

> Mocha will strive to evolve with the needs of the community, but will make every effort not to abandon its current users in doing so.

This statement is the opposite of growth. We are looking backwards into the past, eagerly trying not to hurt anybody. We are favouring the past against the future. We haven't taken any larger decision since several months, the best way to keep the past alive.

I'm not really taking this paper seriously, because it will not have any impact in my daily life as a maintainer, but at least above sentence should be twisted:
"Mocha will make every effort to evolve with the needs of the community, but will strive not to abandon its current users in doing so."

**Comment by boneskull** [comment]
We can change the wording there, though it doesn’t seem to be materially different (to me). 

It hasn’t seen much action in the past six months because, frankly, I’ve been busy with other things and haven’t had the time to put in.  It does not mean the project is dying. The biggest flurry of activity on this project happened when TJ wrote it. If you look at my contributions over the years, you’ll see that my activity comes and goes with significant breaks... yet here we are. 

“Growth” means that we intend to work towards a more mature model of project governance and sustainability. It does not mean we necessarily intend to add a bunch of new features. To that end, I think it’s appropriate for where we’re at.

Mocha has a demonstrated impact in the ecosystem. A lot of people and dev teams and companies rely on Mocha. More projects depend on Mocha every month. As a line graph over time, it’d point up and to the right. 

Insofar as “past it’s prime,” no, nobody gets excited by Mocha. That’s where we’re at. Every project plateaus in this way. 

No, this document won’t really have much of an effect on any of us in terms of technical leadership. It’s not really _for_ us—it’s for the Foundation. Still, it’s helpful to be able to point to this and say why we’re not willing to e.g. ship built-in assertions.

Either way we all need to be comfortable with what this document is saying. I’m happy to have a discussion about the Foundation or project future in another issue (admin repo), Gitter, or otherwise.

**Comment by craigtaub** [comment]
👍 LGTM 

Im fine with the sentence, given as a mission statement its somewhat ambitious and ambiguous anyway.

> Means we intent to work towards a more mature model of project governance and sustainability

Sounds great. I think we need a conversation about what that actually looks like (just LTS?). At this point im not sure.

**Comment by boneskull** [comment]
> I think we need a conversation about what that actually looks like (just LTS?)

Right, that's one part, but we also need to formalize a few things around governance.

File: PROJECT_CHARTER.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+## §3: Relationship with OpenJS Foundation CPC
+
+Technical leadership for the projects within the [OpenJS Foundation] is delegated to the projects through their project charters by the [OpenJS Foundation Cross-Project Council](https://openjsf.org/about/governance/) (CPC). In the case of the Mocha project, it is delegated exclusively to the maintainers of Mocha. The OpenJS Foundation's business leadership is the Board of Directors.
+
+Changes to the following **cannot** unilaterally be applied by project leadership, and must be ratified by the CPC:
+
+- Mocha's Project Charter (this document)
+- Mocha's [Code of Conduct]
+- Mocha's licenses ([Apache-2.0](https://github.com/mochajs/mocha/blob/master/LICENSE) for code and [CC-BY-4.0](https://github.com/mochajs/mocha/blob/master/docs/LICENSE-CC-BY-4.0) for website)
```
Comment (outsideris) [inline PROJECT_CHARTER.md]: Apache-2.0? not MIT?

Comment (boneskull) [reply PROJECT_CHARTER.md]: ha good catch!!

**Comment by boneskull** [comment]
@mochajs/core Please review or approve this; I've seen some comments, but I'd like explicit approvals

**Comment by boneskull** [comment]
We may need to change it further since it must go through the CPC, but I'll get that process started.

**Comment by boneskull** [comment]
Ref: openjs-foundation/cross-project-council#419

**Pull request merged**